### PR TITLE
Libingester added support for gzipped responses

### PR DESCRIPTION
**Pull request by sgnn7**
**DO NOT MERGE BEFORE https://github.com/endlessm/libingester/pull/10** (test code file reorg is in it)

https://phabricator.endlessm.com/T16672

### CODE REVIEW

**Comment by cosimoc** [comment]
@sgnn7 Can you rebase this branch now that #10 is merged?

**Comment by sgnn7** [comment]
@cosimoc I don't see any conflicts - the note was there just because the branch was built on top of the previous one so it should be good.

**Comment by cosimoc** [comment]
@sgnn7 It would be easier to review it, since most of these commits have been merged already :)

**Comment by sgnn7** [comment]
k - let me do that right quick

**Comment by sgnn7** [comment]
@cosimoc done

File: test/lib/util.test.js [hunk 1]
```diff
@@ -68,3 +68,17 @@ describe('download_image', function() {
         expect(asset._content_type).to.equal('image/png');
     });
 });
+
+describe('fetch_html', function() {
+    it('can handle gzipped responses', function(done) {
+        const test_url = 'https://www.kapanlagi.com/' +
+                        'intermezzone/' +
+                        'bule-amerika-ini-nyoba-makan-buah-duku-ekspresinya-nggak-nahan-aee243.html'
```
Comment (cosimoc) [inline test/lib/util.test.js]: Missing semicolon

File: test/lib/util.test.js [hunk 2]
```diff
@@ -68,3 +68,17 @@ describe('download_image', function() {
...
+        const doctype = '<!DOCTYPE html>'
```
Comment (cosimoc) [inline test/lib/util.test.js]: Missing semicolon

File: test/lib/util.test.js [hunk 3]
```diff
@@ -68,3 +68,17 @@ describe('download_image', function() {
...
+        util.fetch_html(test_url).then((result) => {
+            expect(result.html().substring(0, doctype.length)).to.equal(doctype)
```
Comment (cosimoc) [inline test/lib/util.test.js]: Missing semicolon

File: test/lib/util.test.js [hunk 4]
```diff
@@ -68,3 +68,17 @@ describe('download_image', function() {
...
+            done()
```
Comment (cosimoc) [inline test/lib/util.test.js]: Missing semicolon

**Comment by sgnn7** [comment]
@cosimoc Fixed. CR ready again.

**Comment by cosimoc** [comment]
Looks good to me (it's a one-liner after all!) but I'd rather have @wgreenberg's blessing before merging :)

**Comment by wgreenberg** [comment]
oops. CR pass!

**Pull request merged**